### PR TITLE
add params check when convert BatchNorm to GroupNorm

### PR DESCRIPTION
**Pull request by XiaobingSuper**
This PR will do params check when converting BatchNorm to GroupNorm, because GroupNorm should have some pre-request at its initiation step.

### CODE REVIEW

**Comment by JohnlNguyen** [comment]
Hey, these still fail our unit tests
https://app.circleci.com/pipelines/github/pytorch/opacus/1709/workflows/446de3e4-f2c7-4090-ba3f-7c8793cb680f/jobs/9174

**Comment by XiaobingSuper** [comment]
@JohnlNguyen , the failed case try to convert **BatchNorm2d(72, eps=0.001, momentum=0.01, affine=True, track_running_stats=True)** to GroupNorm, but it is invalid，how do we ignore this conversion?

**Review by karthikprasad — CHANGES_REQUESTED** [review]

File: opacus/validators/batch_norm.py [hunk 1]
```diff
@@ -85,6 +85,10 @@ def _batchnorm_to_groupnorm(module: BATCHNORM) -> nn.GroupNorm:
         A default value of 32 is chosen for the number of groups based on the
         paper *Group Normalization* https://arxiv.org/abs/1803.08494
     """
+    if module.num_features % min(32, module.num_features) != 0:
+       raise UnsupportableModuleError(
+                "There is no equivalent GroupNorm  module to replace BatchNorm with."
+            )
     return nn.GroupNorm(
         min(32, module.num_features), module.num_features, affine=module.affine
```
[the remark this answers is not in the document]
Comment (XiaobingSuper) [reply opacus/validators/batch_norm.py]: Thanks, the code is changed to use the **gcd** method.

**Review by karthikprasad — APPROVED** [review]

File: opacus/validators/batch_norm.py [hunk 2]
```diff
@@ -85,6 +85,10 @@ def _batchnorm_to_groupnorm(module: BATCHNORM) -> nn.GroupNorm:
         A default value of 32 is chosen for the number of groups based on the
         paper *Group Normalization* https://arxiv.org/abs/1803.08494
     """
+    if module.num_features % min(32, module.num_features) != 0:
+       raise UnsupportableModuleError(
+                "There is no equivalent GroupNorm  module to replace BatchNorm with."
+            )
```
[the remark this answers is not in the document]
Comment (gchanan) [reply opacus/validators/batch_norm.py]: what was the previous behavior?  You would get an error message if and only if you actually ran the module?

[the remark this answers is not in the document]
Comment (XiaobingSuper) [reply opacus/validators/batch_norm.py]: @gchanan, yes, you only get an error when running the module before.
